Documentation/contributing.rst

```diff
@@ -565,6 +582,27 @@ Best Practices for Writing Tests
         vm.ReportFailed()
     }
 
+
+Ginkgo Extensions
+~~~~~~~~~~~~~~~~~~
+
+In cilium, some Ginkgo features are extended to cover some uses cases that are
+useful for how cilium run test
+
+BeforeAll
+^^^^^^^^^^
+
+This function will run before all BeforeEach within a describe or context. This
+function is the new ``SetUp`` or initalize function for our Ginkgo test.
+
```

Comment:
Would be useful to provide an example in the Ginkgo extensions section to show what these constructs do in relation to each other (i.e., a dummy test with multiple `It`s, a `BeforeAll`, `AfterAll`, `AfterEach`, `BeforeEach`, etc. to show users in what order the various things run, like;


```Describe --> BeforeAll --> BeforeEach --> It --> AfterEach --> BeforeEach --> It --> AfterEach --> AfterAll```